### PR TITLE
GH-103944: Check error status when raising DeprecationWarning

### PR DESCRIPTION
**Pull request by pganssle**
Fix for #103944.

* Issue: gh-103944

### CODE REVIEW

File: Modules/_datetimemodule.c [hunk 1]
```diff
@@ -5144,13 +5144,12 @@ datetime_datetime_now_impl(PyTypeObject *type, PyObject *tz)
...
-        PyExc_DeprecationWarning,
-        "datetime.utcnow() is deprecated and scheduled for removal in a future "
-        "version. Use timezone-aware objects to represent datetimes in UTC: "
-        "datetime.now(datetime.UTC).",
-        2
-    );
+    if(PyErr_WarnEx(PyExc_DeprecationWarning,
+        "datetime.utcnow() is deprecated and scheduled for removal in a "
+        "future version. Use timezone-aware objects to represent datetimes "
+        "in UTC: datetime.now(datetime.UTC).", 2)) {
```
Comment (erlend-aasland) [inline Modules/_datetimemodule.c]: PEP-7 nit

```suggestion
        "in UTC: datetime.now(datetime.UTC).", 2))
    {
```

File: Modules/_datetimemodule.c [hunk 2]
```diff
@@ -5187,13 +5186,12 @@ datetime_fromtimestamp(PyObject *cls, PyObject *args, PyObject *kw)
...
 {
-    PyErr_WarnEx(
-        PyExc_DeprecationWarning,
+    if(PyErr_WarnEx(PyExc_DeprecationWarning,
         "datetime.utcfromtimestamp() is deprecated and scheduled for removal "
         "in a future version. Use timezone-aware objects to represent "
-        "datetimes in UTC: datetime.now(datetime.UTC).",
-        2
-    );
+        "datetimes in UTC: datetime.now(datetime.UTC).", 2)) {
```
Comment (erlend-aasland) [inline Modules/_datetimemodule.c]:
```suggestion
        "datetimes in UTC: datetime.now(datetime.UTC).", 2))
    {
```

File: Modules/_datetimemodule.c [hunk 3]
```diff
@@ -5144,13 +5144,12 @@ datetime_datetime_now_impl(PyTypeObject *type, PyObject *tz)
...
 datetime_utcnow(PyObject *cls, PyObject *dummy)
 {
-    PyErr_WarnEx(
-        PyExc_DeprecationWarning,
-        "datetime.utcnow() is deprecated and scheduled for removal in a future "
-        "version. Use timezone-aware objects to represent datetimes in UTC: "
-        "datetime.now(datetime.UTC).",
-        2
-    );
+    if(PyErr_WarnEx(PyExc_DeprecationWarning,
```
Comment (erlend-aasland) [inline Modules/_datetimemodule.c]: PEP-7 nit

```suggestion
    if (PyErr_WarnEx(PyExc_DeprecationWarning,
```

File: Modules/_datetimemodule.c [hunk 4]
```diff
@@ -5187,13 +5186,12 @@ datetime_fromtimestamp(PyObject *cls, PyObject *args, PyObject *kw)
 static PyObject *
 datetime_utcfromtimestamp(PyObject *cls, PyObject *args)
 {
-    PyErr_WarnEx(
-        PyExc_DeprecationWarning,
+    if(PyErr_WarnEx(PyExc_DeprecationWarning,
```
Comment (erlend-aasland) [inline Modules/_datetimemodule.c]: PEP-7 nit

```suggestion
    if (PyErr_WarnEx(PyExc_DeprecationWarning,
```

**Review by erlend-aasland — COMMENTED** [review]
I left some PEP-7 nits.

**Comment by pganssle** [comment]
@erlend-aasland OK, I've made the relevant changes. Thanks! (My kingdom for an incremental PEP 7 formatter... 😅)

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**